web/src/components/Header.test.js

```diff
@@ -0,0 +1,117 @@
+import { shallow, mount, debug } from 'enzyme';
+import React from 'react';
+import sinon from 'sinon';
+import { push } from 'connected-react-router';
+import { Link, MemoryRouter } from 'react-router-dom';
+
+import { plain as Header, mapStateToProps, mapDispatchToProps } from './Header';
+
+describe('Header component', () => {
+  it('renders correctly when the dropdown is closed and the page is top level', ()=> {
+    const component = shallow(
+      <Header
+        authenticated
+        currentUser={{ role: "admin", state: { id: "wa", name: "Washington" }, username: "frasiercrane@kacl.com"}}
+        isAdmin
+        pushRoute={() => {}}
+        ariaExpanded={false}
+        showSiteTitle
+      />
+    );
+    expect(component).toMatchSnapshot();
+  });
+
+  it('opens and closes the dropdown when clicked', () => {
+    const component = shallow(
+      <Header
+        authenticated
+        currentUser={{ role: "admin", state: { id: "wa", name: "Washington" }, username: "frasiercrane@kacl.com"}}
+        isAdmin
+        pushRoute={() => {}}
+        ariaExpanded={false}
+        showSiteTitle
+      />
+    );
+    component.find('.nav--dropdown__trigger').simulate('click');
+    expect(component).toMatchSnapshot();
+    expect(component.state().ariaExpanded).toBeTruthy();
+
+    component.find('.nav--dropdown__trigger').simulate('click');
+    expect(component.state().ariaExpanded).toBeFalsy();
+  });
+
+  it('renders the admin home title when an admin user is at a secondary page', () => {
+    const component = mount(
```

Comment:
In order to get the link text of the `Link` element, I had to mount the whole dang thing. Which required `MemoryRouter`, for some reason I don't pretend to understand.

Reply:
![well duh](https://media1.tenor.com/images/1f943c7d49ca3aaf3b62fb0de4c7c4e3/tenor.gif)